src/MessageReader.js

```diff
@@ -39,22 +39,23 @@ class StandardTypeReader {
     this.view = new DataView(buffer.buffer, buffer.byteOffset);
   }
 
-  string() {
+  string(isJson: boolean) {
     const len = this.int32();
     const codePoints = new Uint8Array(this.buffer.buffer, this.buffer.byteOffset + this.offset, len);
     this.offset += len;
     // if the string is relatively short we can use apply
     // but very long strings can cause a stack overflow due to too many arguments
     // in those cases revert to a slower itterative string building approach
     if (codePoints.length < 1000) {
-      return String.fromCharCode.apply(null, codePoints);
+      const resultString = String.fromCharCode.apply(null, codePoints);
+      return isJson ? JSON.parse(resultString) : resultString;
     }
 
     let data = "";
     for (let i = 0; i < len; i++) {
       data += String.fromCharCode(codePoints[i]);
     }
-    return data;
+    return isJson ? JSON.parse(data) : data;
```

Comment:
What if the parsing fails? Should we just return `undefined` in that case? Or a string with the error message in it maybe?

Reply:
Will return string like `Could not parse ${data}`